### PR TITLE
Zeckert/create sagas for o auth form

### PR DESCRIPTION
**Pull request by zeckertG**
Updating OAuth Form to use sagas

### CODE REVIEW

File: packages/extension-utils/src/OAuthForm/src/state/sagas.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    )
+  } catch (error: any) {
+    yield* put(handleVerifyActionFailure(error.message))
+  }
+}
+
+export function* saga() {
+  const {
+    initAction,
+    handleUrlChangeAction,
```
Comment (zeckertG) [inline packages/extension-utils/src/OAuthForm/src/state/sagas.ts:191]: Should I drop the 'handle' from these actions?

File: packages/extension-utils/src/OAuthForm/src/components/OAuthForm.tsx [hunk 2]
```diff
@@ -113,88 +105,31 @@ export const OAuthForm = ({
...
-        message: `'${value}' is not a valid url`,
-        type: 'error',
-      }
-    }
-
-    updateApiServerUrlAction({
-      apiServerUrl: newApiServerUrl,
-      webUrl: '',
-      validationMessages: newValidationMessages as ValidationMessages,
+    handleUrlChangeAction({
```
Comment (zeckertG) [inline packages/extension-utils/src/OAuthForm/src/components/OAuthForm.tsx:130]: Is this something I could do inline the JSX, or is it better to leave it out of it?

File: packages/extension-utils/src/OAuthForm/src/state/sagas.ts (hunk 1 above)
Comment (josephaxisa) [reply packages/extension-utils/src/OAuthForm/src/state/sagas.ts:191]: you could change them to `setUrlAction`, `verifyAction`, `saveConfigAction`, `clearConfigAction` respectively. The handlers themselves should follow the naming convention `handle*`.

File: packages/extension-utils/src/OAuthForm/src/components/OAuthForm.tsx (hunk 2 above)
Comment (josephaxisa) [reply packages/extension-utils/src/OAuthForm/src/components/OAuthForm.tsx:130]: you could but I'd recommend against it as it makes debugging harder

File: packages/extension-utils/src/OAuthForm/src/state/sagas.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    clearFormAction,
+    handleVerifyAction,
+    handleSaveConfigAction,
+  } = OAuthFormActions
+  yield* takeEvery(initAction, initSaga)
+  yield* takeEvery(handleUrlChangeAction, handleUrlChangeSaga)
+  yield* takeEvery(clearFormAction, clearFormSaga)
+  yield* takeEvery(handleVerifyAction, handleVerifySaga)
+  yield* takeEvery(handleSaveConfigAction, handleSaveConfigSaga)
+}
```
Comment (josephaxisa) [inline packages/extension-utils/src/OAuthForm/src/state/sagas.ts:201]: Let's add full test coverage for all the sagas in a `sagas.spec.ts` file

**Pull request closed without merging**